src/ImageSharp/ColorProfiles/Cmyk.cs

```diff
@@ -157,8 +157,5 @@ public override bool Equals(object? obj)
     /// <inheritdoc/>
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public bool Equals(Cmyk other)
-        => this.C.Equals(other.C)
-        && this.M.Equals(other.M)
-        && this.Y.Equals(other.Y)
-        && this.K.Equals(other.K);
+        => new Vector4(this.C, this.M, this.Y, this.K) == new Vector4(other.C, other.M, other.Y, other.K);
```

Comment:
Another idea: see [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgDkBXfGKASzFwBuGvSasAShwB2GXtxYBhCPgAOvADY8AyjwBu/GEJGNmLSTLkwWASRk8IKnVH1hDw6iIDMDWNgAmEFLqAJ4MuBhQHGAYDAr4wQDWNADeNAzpTN4AZuoQ2DEKDMkMAOYwGIIMAL5pGcTZufkMALJFpeWVNdQZmQw5eTEAmm1lFdW16fV9jTEA0iMd490ZEwwA2s3lABYQftaq6gAUmxg7ewcA8iqygbgsAIIlJbC4uLy6MLbqvFI/JQCUAF1VlNgBAIOoGABRACOHGw6lwhziiQYEFOPH+qx6AF4AHwMKQwADuDAAajBotAUIdTrw7go0Aw6XdmkyWSxBuytvSWLN/gwcTjCSTyZSMNTDuitjxFEzpbK2WiMVBOfKVXz/u4ujQ1loIlEMAAZbDBCAcDCHE1mi2zH5+Fg6OEwCwIoFeHwwfyBEJhA3RWLxBIAfSo1FSy0mDQGsQWYy6PSm/SarWKo06IOjTWGacWCbqWbmcYzkYYqxUfF0+RgYqpUBQDHuuApdYb+IYAFUpLhsFkrE2ADwokPkJktiX1vGHWBZTvd3v93DiGBZQ4/Zk83D/LWrVYbba7fYqI4nM5H9RXG7dh5PF5vD5fH5/d2l0HgyGw+GI5FB0PKmVQFipa4gSHJNuO1KCsKCqquB4rUtqQA==) with the private `AsVector4` property.
This overlays the vector with the struct (is OK for float, as there's no padding in between), so the creation of the vector can be avoided.

That should work for other places (Vector3) too.

Reply:
Oh that's cleaver yeah!